cpp/src/slice2py/PythonUtil.cpp

```diff
@@ -2872,6 +3047,36 @@ Slice::Python::validatePythonMetadata(const UnitPtr& unit)
                        "' metadata can only be applied to sequences of bools, bytes, shorts, ints, longs, floats, "
                        "or doubles";
             }
+
+            if (m->directive() == "python:memoryview")
+            {
+                // Argument can't be empty as it sets MetadataArgumentKind::RequiredTextArgument
+                auto arguments = m->arguments();
+                assert(!arguments.empty());
```

Comment:
The code-gen will never see bad identifiers, but this is run in the validator, before we can make any assumptions about whether or not the metadata is valid.

So it's possible that the arguments will be empty at this point.
Another part of the validator will emit an error for it of course, but that check could happen _after_ this one.

Reply:
I removed the assert.